### ISSUE TITLE
Webdav backend incompatible with golang stdlib webdav implementation

### ISSUE THREAD

**Issue by SISheogorath** [stevesoltys/backup#825]
I think I found conflict between how the seedvault implementation expects directories to work and how golang handles it:

In this function, you test that directories exist using the HEAD function:
https://github.com/seedvault-app/seedvault/blob/156ffbd8bc7d8e9f35472454fb00cb6c0a167aee/core/src/main/java/org/calyxos/seedvault/core/backends/webdav/WebDavHelper.kt#L84-L103

golang doesn't support HEAD requests on directory listings:
https://github.com/golang/net/blob/dfc720dfe0cfc125116068c20efcdcb5e4eab464/webdav/webdav.go#L212

I checked the RFCs [RFC4918](http://www.webdav.org/specs/rfc4918.html#rfc.section.9.4) and [RFC2068](https://datatracker.ietf.org/doc/html/rfc2068#section-9.4) but couldn't find any defintive answer how this should be handled. But there isn't really an answer to this, beyond "Treat GET and HEAD identical". The overall internet also seems undecided how to handle directory requests in webdav. ([Here one of the samples I found](https://stackoverflow.com/a/24338433)).

I think the correct way to check for directories would be the usage of `PROPFIND` with a depth of 0. That would certainly be in line with the [RFC4918 section 9.1](http://www.webdav.org/specs/rfc4918.html#rfc.section.9.1).

This is an upstream issue for downstream issue https://github.com/GrapheneOS/os-issue-tracker/issues/4558

**Comment by grote** [comment]
shouldn't be too difficult to change the head request to propfind(depth=0).

**Comment by grote** [comment]
@martijnboers are you using https://github.com/mholt/caddy-webdav ? If so, it seems that also doesn't support `HEAD` requests. I am getting `405` from it.

**Comment by SISheogorath** [comment]
It's based on the same library: https://github.com/mholt/caddy-webdav/blob/42168ba04c9dc2cd228ab8c453dbab27654e52e6/webdav.go#L30

**Comment by noisesfromspace** [comment]
Yes I am using that caddy extension, I can setup a non-Go implementation for debugging if that helps. I was looking through the forks but it seems no one has bothered implementing `HEAD` requests.

**Comment by grote** [comment]
I am going to submit a PR that removes the HEAD request in favor of PROPFIND. With that it works fine against your caddy server and much much faster than going through SAF and DavX5.
